src/wasm-emscripten.cpp

```diff
@@ -196,6 +196,96 @@ void EmscriptenGlueGenerator::generateDynCallThunks() {
   }
 }
 
+struct JSCallWalker : public PostWalker<JSCallWalker> {
+  Module &wasm;
+  JSCallWalker(Module &_wasm) : wasm(_wasm) {
+    if (wasm.table.segments.size() == 0) {
+      auto emptySegment =
+          wasm.allocator.alloc<Const>()->set(Literal(uint32_t(0)));
+      wasm.table.segments.emplace_back(emptySegment);
+    }
+    std::vector<Name> tableSegmentData = wasm.table.segments[0].data;
```

Comment:
Can be `auto`.

Reply:
Maybe const auto& would be better to prevent copying?